packages/react-instantsearch-dom/src/components/SmartSort.tsx

```diff
@@ -13,23 +13,29 @@ export type SmartSortProps = {
   className?: string;
   isVirtualReplica: boolean;
   isSmartSorted: boolean;
-  buttonTextComponent: React.FunctionComponent<SmartSortComponentProps>;
-  textComponent: React.FunctionComponent<SmartSortComponentProps>;
+  buttonTextComponent?: React.FunctionComponent<SmartSortComponentProps>;
+  textComponent?: React.FunctionComponent<SmartSortComponentProps>;
   refine(relevancyStrictness: number | undefined): void;
 };
 
+const DefaultButtonTextComponent = ({
+  isSmartSorted,
+}: SmartSortComponentProps) => (
+  <div>{isSmartSorted ? 'See all results' : 'See relevant results'}</div>
```

Comment:
```suggestion
  <span>{isSmartSorted ? 'See all results' : 'See relevant results'}</span>
```

I think we can still change it to be a `span`.

Reply:
Sure!